### ISSUE TITLE
question   how can i fix dbus error

### ISSUE THREAD

**Issue by djdisodo** [CypherpunkArmory/UserLAnd#157]
i tried to reconfigure dbus first

`sudo dpkg-reconfigure dbus`

and rebooted my phone
(reconfigure required reboot but sudo reboot is not worked )

`sudo DISPLAY=:0 PULSE_SERVER=tcp:127.0.0.1:4712 startxfce4`

it outputs

`djdisodo@localhost:~$ sudo startxfce4
/usr/bin/startxfce4: X server already running on display :0

(xfce4-session:18039): xfce4-session-CRITICAL **: 17:47:46.482: Unable to contact D-Bus session bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.

(xfce4-session:18039): xfce4-session-CRITICAL **: 17:47:46.778: Unable to contact D-Bus session bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.789: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.793: dbus_g_proxy_connect_signal: assertion 'DBUS_IS_G_PROXY (proxy)' failed

** (xfce4-session:18039): CRITICAL **: 17:47:46.795: dbus_g_proxy_connect_signal: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.797: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.799: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:46.803: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.806: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.808: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:46.817: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.847: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.851: dbus_g_proxy_connect_signal: assertion 'DBUS_IS_G_PROXY (proxy)' failed

** (xfce4-session:18039): CRITICAL **: 17:47:46.853: dbus_g_proxy_connect_signal: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.856: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.858: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:46.860: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.906: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.908: dbus_g_proxy_disconnect_signal: assertion 'DBUS_IS_G_PROXY (proxy)' failed

** (xfce4-session:18039): CRITICAL **: 17:47:46.911: dbus_g_proxy_disconnect_signal: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:46.917: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:46.919: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:46.921: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.216: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.218: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.220: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.230: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.233: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.235: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.236: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.238: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.240: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.250: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.252: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.255: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.256: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.258: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.260: g_propagate_error: assertion 'src != NULL' failed
xfce4-session: No SSH authentication agent found
gpg-agent[18062]: WARNING: "--write-env-file" is an obsolete option - it has no effect
gpg-agent: a gpg-agent is already running - not starting a new one

(xfce4-session:18039): xfce4-session-WARNING **: 17:47:47.617: gpg-agent returned no PID in the variables

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.618: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.620: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.622: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.623: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.625: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.626: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.628: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.631: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.632: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.642: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.646: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.648: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.650: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.653: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.655: g_propagate_error: assertion 'src != NULL' failed

(xfce4-session:18039): xfce4-session-WARNING **: 17:47:47.658: xfsm_manager_load_session: Something wrong with /home/djdisodo/.cache/sessions/xfce4-session-localhost:0, Does it exist? Permissions issue?

(xfce4-session:18039): xfconf-CRITICAL **: 17:47:47.661: xfconf_init() must be called before attempting to use libxfconf!

** (xfce4-session:18039): CRITICAL **: 17:47:47.663: dbus_g_proxy_call: assertion 'DBUS_IS_G_PROXY (proxy)' failed

(xfce4-session:18039): GLib-CRITICAL **: 17:47:47.665: g_propagate_error: assertion 'src != NULL' failed
^C
djdisodo@localhost:~$`

on xsdl
> unable to contact settings server

**Comment by AdamMatthewLTM** [comment]
@djdisodo ,

Thank you for filing this issue. There are some known issues when attempting to run xfce and known issues with dbus that are being worked on at the moment. Before I can provide a potential fix however, could you please let me know what Android version and phone model you are running and any other important information not included in the template?

Thank you for your consideration,
-Adam

**Comment by djdisodo** [comment]
@AdamMatthewLTM 
![screenshot_20180807-132722](https://user-images.githubusercontent.com/17721656/43754435-5e360e02-99fa-11e8-88a0-6b24499cb412.png)
![screenshot_20180807-132726](https://user-images.githubusercontent.com/17721656/43754452-6aaf47e8-99fa-11e8-9d0a-a6ef66cd7ef5.png)

it has same error in GNUroot Debian

**Comment by corbinlc** [comment]
Can people confirm if they are still having an issue with this?  I am having no problem running xfce or lxde after the last proot update. I am not sure about KDE as I have not tried it.

**Comment by JanuszChmiel** [comment]
D-BUS issue exist unfortunately till now. It will be very complex to completely solve them.
To debug it, try to run
sudo service dbus start
Sure, You must install D-bus service previously by install its package.

When you will this service as user not as system, since it is impossible to run system service by using command to run this kind of service in Linux.
But when D-BUS run by using
sudo service dbus start
There are other errors related to D-bus configuration. Those errors will persists even when you will try to install some packages such as LXDE.
LXDE produces no errors while running, but I Am experiencing other problem. Even when I press back key on The XServer XSDL screen, so keyboard is active inside X session, LXDE can not detect hod key presses. Keyboard is detected with no issues, when I run firefox without running whole desktop. I simply start firefox by typing firefox from terminal. In this case, Firefox detect even hod keys and individual letters. I can even load whole page and system is stable, produces no crashes.
Gnome and \Mate can not be used, since Gnome-keyring report issues related to process, "insufficient process capabilities" and some other bug related to memory protected. May be, that Android allocates too many MB of RAM and process do not have enough of allocatable memory to use it.
May be, that it is some kind of errors while D-BUS do not work as it should.
I Am running Lenovo TAB 38, with latest Android 6.0 update from 9 month of last year.
And Lenovo enabled device encryption by default and there is no way to turn it 
[PRoot_Debug_Log .txt](https://github.com/CypherpunkArmory/UserLAnd/files/2396584/PRoot_Debug_Log.txt)
off. So may be, that this encryption algorithm makes CPU of my tablet more overloaded by encryption and decryption in real time. I Am attaching Proot_Debug_log.txt. I have set highest level of debugging inside Userland settings dialog window. I have used Debian, I have tried to run Lxde desktop.
